### PR TITLE
Add org.dupot.easyflatpak

### PR DESCRIPTION
**Pull request by imikado**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  
Application to simplify installation of some flatpak 
- [x ] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [ x] I have read the and followed all the [Submission and licence requirements][reqs].
- [ x] I have [built][build] and tested the submission locally.
- [ x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: org.dupot.easyflatpak.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+app-id: org.dupot.easyflatpak
+runtime: org.gnome.Platform
+runtime-version: "45"
```
Comment (hfiguiere) [inline org.dupot.easyflatpak.yml:3]: this runtime will be EOL in less than a couple of month.

Comment (imikado) [reply org.dupot.easyflatpak.yml]: i change to freedesktop

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+"only-arches": ["x86_64"]
```
[the remark this answers is not in the document]
Comment (imikado) [reply flathub.json:2]: i should modify ?

[the remark this answers is not in the document]
Comment (imikado) [reply flathub.json]: i just updated to remove this restriction

[the remark this answers is not in the document]
Comment (hfiguiere) [reply flathub.json]: and how will this work since you don't build it from source?

[the remark this answers is not in the document]
Comment (imikado) [reply flathub.json]: so i rollback to previous

i don't know how to build from source for flutter application

File: org.dupot.easyflatpak.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+separate-locales: false
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --share=ipc
+  - --device=dri
+  - --filesystem=/var/lib/flatpak/app:ro
+  - --filesystem=xdg-data/flatpak/app:ro
+  - --filesystem=xdg-data/flatpak/overrides:create
```
[the remark this answers is not in the document]
Comment (bbhtt) [reply org.dupot.easyflatpak.yml:15]: "Application to simplify installation of some flatpak" - why is access to the second one needed? That is not used when installing/uninstalling.

**Comment by bbhtt** [comment]
bot, build org.dupot.easyflatpak

File: org.dupot.easyflatpak.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (bbhtt) [reply org.dupot.easyflatpak.yml:15]: These are the error codes for the exception request.

```
"finish-args-unnecessary-xdg-data-flatpak-create-access",
"finish-args-unnecessary-xdg-data-flatpak-ro-access"
```

[the remark this answers is not in the document]
Comment (imikado) [reply org.dupot.easyflatpak.yml]: i just removed it

[the remark this answers is not in the document]
Comment (bbhtt) [reply org.dupot.easyflatpak.yml]: Why did you have them in the first place?

File: org.dupot.easyflatpak.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+command: dupot_easy_flatpak
+separate-locales: false
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=ipc
+  - --device=dri
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (bbhtt) [inline org.dupot.easyflatpak.yml:12]: Why is flatpak-spawn required? This is not allowed by default.

Comment (imikado) [reply org.dupot.easyflatpak.yml:12]: my application call flatpak to override permission for some application after their installation 
for example for steam to add game's user folder

File: org.dupot.easyflatpak.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (imikado) [reply org.dupot.easyflatpak.yml]: i thought it was needed but not

File: org.dupot.easyflatpak.yml (hunk 4 above)
Comment (bbhtt) [reply org.dupot.easyflatpak.yml:12]: You said in the description `Application to simplify installation of some flatpak` now you are saying it needs to override permissions.

What exactly is this app?

**Comment by bbhtt** [comment]
Your project on Github seems like a demo application. `pubspec.yaml` contains random placeholder description https://github.com/imikado/dupotEasyFlatpak/blob/dae3ee410d26642a9c5fb92e86f56e0a5b32f1b4/pubspec.yaml#L2

The source code files are named like a hello world program https://github.com/imikado/dupotEasyFlatpak/blob/main/linux/my_application.cc

The readme seems to contain generated placeholder text https://github.com/imikado/dupotEasyFlatpak?tab=readme-ov-file#getting-started

The application ID is com.example https://github.com/imikado/dupotEasyFlatpak/blob/dae3ee410d26642a9c5fb92e86f56e0a5b32f1b4/linux/CMakeLists.txt#L10

File: org.dupot.easyflatpak.yml (hunk 4 above)
Comment (imikado) [reply org.dupot.easyflatpak.yml:12]: i just updated the readme..
The idea is very simple: you have a folder which store recipies for each application in json format.
In each json file, you have title,description, flatpak id + a list of override action needed after installation
For example, for steam, after installation, the application ask used to setup the folder where his games are stored to add it with --filesystem override

**Comment by imikado** [comment]
I can't answer the last comment https://github.com/flathub/flathub/pull/5466#issuecomment-2266348524

for information to use flatpak-spawn, i need to setup something no ? it is avaiable in flatpak container ?

**Comment by bbhtt** [comment]
Detect when it is inside the sandbox and run `flatpak-spawn --host flatpak` instead of `flatpak`

**Comment by imikado** [comment]
Hi
answer to https://github.com/flathub/flathub/pull/5466#issuecomment-2267235160
i push a new version using flatpak-spawn
For information i think i have a problem with my fullter setup, if i try to launch my app on my computure (kubuntu 24.04), it works, it work on a fresh ubuntu install (in vm), but it does not work on debian 12.6
i wll test to build the project from a vm running on debian to check 
i don't know if the application start on your test environment

**Comment by imikado** [comment]
my application does not found flatpak-spawn
i got this error

"Unhandled Exception: ProcessException: No such file or directory
  Command: flatpak-spawn --host flatpak info com.discordapp.Discord"

**Comment by bbhtt** [comment]
Flatpak spawn does not work outside the flatpak sandbox.

File: org.dupot.easyflatpak.yml (hunk 4 above)
Comment (bbhtt) [reply org.dupot.easyflatpak.yml:13]: can you open a PR for the exception? https://docs.flathub.org/docs/for-app-authors/linter#exceptions

This is the error code https://docs.flathub.org/docs/for-app-authors/linter#finish-args-flatpak-spawn-access

Comment (imikado) [reply org.dupot.easyflatpak.yml:13]: Thanks for advice
i did it : https://github.com/imikado/flatpak-builder-lint/pull/1
Should i click on "merge pull request" button ?

Comment (imikado) [reply org.dupot.easyflatpak.yml:13]: i clicked on merge pull request

Should i do other modifications ?

Comment (bbhtt) [reply org.dupot.easyflatpak.yml:13]: Don't merge, click on Open pull request. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request#creating-the-pull-request

File: org.dupot.easyflatpak.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+sdk: org.freedesktop.Sdk
+command: dupot_easy_flatpak
+separate-locales: false
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=ipc
+  - --device=dri
+
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (bbhtt) [inline org.dupot.easyflatpak.yml:13]:
```suggestion
  - --device=dri
  - --talk-name=org.freedesktop.Flatpak
```

File: org.dupot.easyflatpak.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      - ln -s /app/EasyFlatpak/dupot_easy_flatpak /app/bin/dupot_easy_flatpak
+      - install -Dm644 git_repo/export/flatpak/${FLATPAK_ID}.appdata.xml /app/share/appdata/${FLATPAK_ID}.appdata.xml
+      - install -Dm644 git_repo/export/flatpak/512x512.png /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm644 git_repo/export/flatpak/${FLATPAK_ID}.desktop /app/share/applications/${FLATPAK_ID}.desktop
+
+    sources:
+      #- type: archive
+      #  dest: EasyFlatpakArchive
+      #  path: bundle.tar.gz
+      #  sha256: 9b2c86d867514002cafff2a716a221d0e46ea5969681b3b32b02f5c4e9b0eb9f
```
Comment (bbhtt) [inline org.dupot.easyflatpak.yml:31]:
```suggestion
```

**Comment by bbhtt** [comment]
bot, build org.dupot.easyflatpak

File: org.dupot.easyflatpak.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
+app-id: org.dupot.easyflatpak
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+command: dupot_easy_flatpak
+separate-locales: false
```
[the remark this answers is not in the document]
Comment (imikado) [reply org.dupot.easyflatpak.yml]: i follow a example for this file with an author compiling a flutter application like me
i accepted your suggesion

**Comment by bbhtt** [comment]
English localisation is missing in various places 

The settings button in the corner seems to do nothing? 

image

**Comment by imikado** [comment]
> English localisation is missing in various places
> 
> The settings button in the corner seems to do nothing?
> 
> image

I remove the settings useless button
i translate summary and description

**Comment by bbhtt** [comment]
bot, build org.dupot.easyflatpak

**Comment by bbhtt** [comment]
Where are these recipes supposed to be installed https://github.com/imikado/dupotEasyFlatpak/?tab=readme-ov-file#create-a-recipie ?

"Create file assets/recipies/heroic.json" <- that's a directory in your git repo, where should I place them once the app is installed?

**Comment by imikado** [comment]
> English localisation missing here
> 
> image
> 
> image
> 
> image
> 
> image

i just push a new version with localization

**Comment by imikado** [comment]
> Where are these recipes supposed to be installed https://github.com/imikado/dupotEasyFlatpak/?tab=readme-ov-file#create-a-recipie ?
> 
> "Create file assets/recipies/heroic.json" <- that's a directory in your git repo, where should I place them once the app is installed?

Recipies are stored in the project source (opensource project) and is compiled by flutter when create a new version of the application, not avaiable for final user when they install the application
This readme is for developper who want to participate

**Comment by bbhtt** [comment]
> Recipies are stored in the project source (opensource project) and is compiled by flutter when create a new version of the application, not avaiable for final user when they install the application
This readme is for developper who want to participate

Then the app has too limited functionality to be useful. An app to install 5 Flatpaks with no ability to add custom recipes by the user, isn't really fit to be included at the moment.

The submission can be reconsidered if more functionality is added later on such as the ability to add custom recipes by the user or install an app through Flatpak.

**Comment by imikado** [comment]
> > Recipies are stored in the project source (opensource project) and is compiled by flutter when create a new version of the application, not avaiable for final user when they install the application
> > This readme is for developper who want to participate
> 
> Then the app has too limited functionality to be useful. An app to install 5 Flatpaks with no ability to add custom recipes by the user, isn't really fit to be included at the moment.
> 
> The submission can be reconsidered if more functionality is added later on such as the ability to add custom recipes by the user or install an app through Flatpak.

I will add this possibility for user to add his own custom recipies

**Comment by imikado** [comment]
> > Recipies are stored in the project source (opensource project) and is compiled by flutter when create a new version of the application, not avaiable for final user when they install the application
> > This readme is for developper who want to participate
> 
> Then the app has too limited functionality to be useful. An app to install 5 Flatpaks with no ability to add custom recipes by the user, isn't really fit to be included at the moment.
> 
> The submission can be reconsidered if more functionality is added later on such as the ability to add custom recipes by the user or install an app through Flatpak.

That's ok: i just push a new version with a button on top right to create a recipe
the format is in the readme https://github.com/imikado/dupotEasyFlatpak

**Comment by imikado** [comment]
I pushed a new version with the feature asked (allow user to add his own recipe), is it ok to you ?
Need i to make other modifications ?

**Comment by bbhtt** [comment]
bot, build org.dupot.easyflatpak

**Comment by bbhtt** [comment]
Seems to work, thanks.

One issue is since, the filesystem permission to override is being asked through a filechooser, and because you don't have any filesystem access, it is getting a path from the portal.

So the override looks like this:

```
[Context]
filesystems=/run/user/1000/doc/3619edb8/Downloads;
```

when it should be:

```
[Context]
filesystems=xdg-download;
```

or, `home/Downloads` or `~/Downloads` etc.

**Comment by bbhtt** [comment]
Instead of going through the filechooser you should probably show a dialogue box to type in the override directly and just write that.

**Comment by imikado** [comment]
> Instead of going through the filechooser you should probably show a dialogue box to type in the override directly and just write that.

i can add 
  - --filesystem=host


i tested localy it works

**Comment by bbhtt** [comment]
The problem is `host` doesn't cover every filesystem override the user can give. It's fine for now I guess.

bot, build org.dupot.easyflatpak

**Comment by imikado** [comment]
> The problem is `host` doesn't cover every filesystem override the user can set. It's fine for now I guess.
> 
> bot, build org.dupot.easyflatpak

it does not work with other partition or usb disk for example ?

i will work on the way you advice after

**Comment by bbhtt** [comment]
No, the following filesystem permissions need to explicitly required  https://docs.flatpak.org/en/latest/sandbox-permissions.html#reserved-paths

> Additionally the following directories from host need to be explicitly requested with --filesystem and are not available with home, host, host-os, host-etc by default:

**Comment by imikado** [comment]
> No, the following filesystem permissions need to explicitly required https://docs.flatpak.org/en/latest/sandbox-permissions.html#reserved-paths
> 
> > Additionally the following directories from host need to be explicitly requested with --filesystem and are not available with home, host, host-os, host-etc by default:

ok i will now modify that
i will rmove this override from my applicaiton --filesytem=host
and i will modify the applicaiton to display a text field to write the explicit path where games are stored

thanks for advice, i will work on this version now

**Comment by bbhtt** [comment]
bot, build org.dupot.easyflatpak

**Comment by bbhtt** [comment]
seems fine now.

**Comment by imikado** [comment]
> seems fine now.
What is the status ? do i have some modification to do ?

**Comment by imikado** [comment]
Hi,
Can i help you to update this application's status ?

**Comment by imikado** [comment]
> seems fine now.

Hi,
How can i help to update this application's status ? 
I aswer the question from  barthalion here https://github.com/flathub/flathub/pull/5466#issuecomment-2284444020 but he does not answer my message :(

**Comment by hfiguiere** [comment]
the word you are looking for is patience. we are not a trillion dollars corporation with an army of reviewers and admins. We are just a handful of volunteers.

**Comment by bbhtt** [comment]
The issue with the app is that it feels a bit of middle ground between the software stores like GNOME Software, KDE Discovery that are installed by default on your distro and Flatpak managers like Flatseal, while not having the polish or the same level of functionality as either of them.

The target userbase of Flathub will have software apps from their distro like GNOME Software or KDE Discovery set up or know how to install via flatpak directly.

In the latter case they should be able to use `flatpak override` directly and in the former case if they are non technical users then pasting a block of JSON to override permissions is already out of their comfort zone.

**Comment by imikado** [comment]
> I think you need to define a clear use case first because the "installing apps" part feels redundant when people can already use their distro software store or flatpak cli directly.
> 
> You can probably refactor it to remove the installation part and make it a "permission manager" like Flatseal.

The idea is very simple, the goal of this application is like "play on linux" which provide an easy way to install some applications with a recipe using right version of wine, setup some tricks for this application.. to allow user to simple clik on each application to install it despite have to install many wine version and setup all of this manually

For this application, it will be the same, some application need to have some override setup after installation, and this applicaiton allow lambda user to simple click on each application available to install then overide permission with cleal label of it without need for each  application to read if they have to override something and what variables..

I can improve interface with the icon of each application like other distro store

If they have to first install application with their distribution, then launch this application to check if there is a recipe for each application they install.. 

For the part to add recipe, as mentionned, in my idea, we have to contribute to the project to add new recipe through the github project , so target is developper profile to contribute
But you ask me to add way for user to add a recipe too.. i can improve this interface not to provide json but use button and other gui control if you think it could be useful, but the goal is to install it after clean install so they will lost this custom user recipe i think.. it is better if want to add recipe to contribute to share thise new recipe with all users no?

**Comment by bbhtt** [comment]
> For the part to add recipe, as mentionned, in my idea, we have to contribute to the project to add new recipe through the github project , so target is developper profile to contribute

Making the recipe private so that only developers can add it reduces usability a bit.

As you suggest I think this is more as a permission manager than a Flatpak app installer.

**Comment by imikado** [comment]
> > For the part to add recipe, as mentionned, in my idea, we have to contribute to the project to add new recipe through the github project , so target is developper profile to contribute
> 
> Making the recipe private so that only developers can add it reduces usability a bit.
> 
> As you suggest I think this is more as a permission manager than a Flatpak app installer.

Ok for recipe i can create a gui to create recipe more friendly than create a json

For recipe, the idea is to do both: create application then override permission, so for the user only need to click on one button in one aplication, not to install then go to flatseal

For information, the idea is not mine, i m involved in a french community of linux gamers, and they don't like each time they install a new distribution ( to find the best distribution to play games on linux) to install applications then install flatseal to override for each application permission after looking for what parameter to override

The idea is like Play on linux : you want to install steam, discord.. you click on one button install, the application know this specificity and ask you if necessary the path where you store games. and for discord add the share of your home (needed but not mentionned only in the github documentation https://github.com/flathub/com.discordapp.Discord/blob/master/README.md )

I can improve application to list all flatpak application if you want, it can be a flatpak  gui store. (some distros/de does not have any store as you know)

**Comment by imikado** [comment]
I worked yesterday on the topic (i found your flatpak repo api), i will propose a new version in few days, i will modify the goal of this project

It will be a flatpak GUI with for some application (like steam, discord..), a "recipe" to help user to override mandatory permission (like game path..)

Thanks for your advices

**Comment by imikado** [comment]
I just finished my first version of a Flatpak Store, i hope you will like it

**Comment by bbhtt** [comment]
bot, build org.dupot.easyflatpak

**Comment by imikado** [comment]
Do you like this new version, or do you think i have to fix something ?

**Pull request closed without merging**